core/src/main/java/org/infinispan/expiration/impl/ClusterExpirationManager.java

```diff
@@ -219,7 +229,14 @@ private boolean pollForCompletion(BlockingQueue<CompletableFuture<?>> queue, lon
             log.tracef("Submitting expiration removal for key %s which had lifespan of %s", toStr(key), lifespan);
          }
          AdvancedCache<K, V> cacheToUse = skipLocking ? cache.withFlags(Flag.SKIP_LOCKING) : cache;
-         CompletableFuture<Void> future = cacheToUse.removeLifespanExpired(key, value, lifespan);
+         CompletableFuture<Void> future;
+         if (transactional) {
+            // Transactional is still blocking - to be removed later
+            future = CompletableFuture.supplyAsync(() -> cacheToUse.removeLifespanExpired(key, value, lifespan), asyncExecutor)
```

Comment:
There is a NullPointer with this fix if the reaper is executed

Reply:
Should be fixed now; added the missing `@Inject`